File: Source/Rio.Models/DataTransferObjects/User/UserDetailedDto.cs
using System;

namespace Rio.Models.DataTransferObjects.User
{
    public class UserDetailedDto
    {
        public int UserID { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }

        public string FullName => $"{FirstName} {LastName}";

        public Guid? UserGuid { get; set; }
        public string Phone { get; set; }
        public string LoginName { get; set; }
        public int RoleID { get; set; }
        public string RoleDisplayName { get; set; }

        public bool? HasActiveTrades { get; set; }
        public int? AcreFeetOfWaterPurchased { get; set; }
        public int? AcreFeetOfWaterSold { get; set; }
    }
}

File: Source/Rio.Web/src/app/shared/models/user/user-detailed-dto.ts
import { AccountSimpleDto } from '../account/account-simple-dto';

export class UserDetailedDto {
    UserID: number;
    UserGuid: string;

    FirstName: string;
    LastName: string;
    FullName: string;

    Email: string;
    Phone: string;
    LoginName: string;

    RoleID: number;
    RoleDisplayName: string;

    HasActiveTrades: boolean;
    AcreFeetOfWaterPurchased: number;
    AcreFeetOfWaterSold: number;
    ReceiveSupportEmails: boolean;
    IsActive: boolean;

    AssociatedAccounts: Array<AccountSimpleDto>

    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

